integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/GCPUtils.java

```diff
@@ -0,0 +1,251 @@
+  private static final String SESSION_ID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-session-id";
+  private static final String SESSION_UUID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-session-uuid";
+  private static final String CLUSTER_UUID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-cluster-uuid";
+  private static final String DATAPROC_REGION_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-region";
+  private static final String DATAPROC_CLASSPATH = "/usr/local/share/google/dataproc/lib";
+  private static final CloseableHttpClient HTTP_CLIENT;
+  private static final String SPARK_YARN_TAGS = "spark.yarn.tags";
+  private static final String SPARK_DRIVER_HOST = "spark.driver.host";
+  private static final String SPARK_APP_ID = "spark.app.id";
+  private static final String SPARK_APP_NAME = "spark.app.name";
+  private static final String SPARK_MASTER = "spark.master";
+  private static final String JOB_ID_PREFIX = "dataproc_job_";
+  private static final String JOB_UUID_PREFIX = "dataproc_uuid_";
+  private static final String METADATA_FLAVOUR = "Metadata-Flavor";
+  private static final String GOOGLE = "Google";
+  private static final String SPARK_DIST_CLASSPATH = "SPARK_DIST_CLASSPATH";
+
+  enum ResourceType {
+    CLUSTER,
+    BATCH,
+    INTERACTIVE,
+    UNKNOWN
+  }
+
+  static {
+    RequestConfig config =
+        RequestConfig.custom()
+            .setConnectTimeout(100)
+            .setConnectionRequestTimeout(100)
+            .setSocketTimeout(100)
+            .build();
+    HTTP_CLIENT = HttpClients.custom().setDefaultRequestConfig(config).build();
+  }
+
+  public static boolean isGCPRuntime() {
+    String sparkDistClasspath = Environment.getEnvironmentVariable(SPARK_DIST_CLASSPATH);
+    return (sparkDistClasspath != null && sparkDistClasspath.contains(DATAPROC_CLASSPATH));
+  }
+
+  public static void setDataprocSpecificFacets(
+      Map<String, Object> dataprocProperties, SparkContext sparkContext) {
+    ResourceType resource = identifyResource(sparkContext);
+
+    switch (resource) {
+      case CLUSTER:
+        getDriverHost(sparkContext).ifPresent(p -> dataprocProperties.put("spark.driver.host", p));
+        getClusterName(sparkContext)
+            .ifPresent(p -> dataprocProperties.put("spark.cluster.name", p));
+        getRegionName().ifPresent(p -> dataprocProperties.put("spark.cluster.region", p));
+        getClusterUUID().ifPresent(p -> dataprocProperties.put("spark.cluster.uuid", p));
+        getGCPJobID(sparkContext).ifPresent(p -> dataprocProperties.put("spark.job.id", p));
+        getGCPJobUUID(sparkContext).ifPresent(p -> dataprocProperties.put("spark.job.uuid", p));
+        break;
+      case BATCH:
+        getRegionName().ifPresent(p -> dataprocProperties.put("spark.batch.region", p));
```

Comment:
I assume endpoints return JSONs and here we put them into string fields. Do you really need whole json content? Wouldn't it make sense to extract exactly the fields you need.

Reply:
These endpoints return a single value (corresponding to the field) and not a JSON.